client/ayon_houdini/plugins/publish/collect_usd_value_clips.py

```diff
@@ -0,0 +1,94 @@
+import os
+import hou
+import clique
+
+import pyblish.api
+
+from ayon_houdini.api import plugin
+
+
+class CollectUSDValueClips(plugin.HoudiniInstancePlugin):
+    """Collect USD value clips that are to be written out for a USD publish.
+
+    It detects sequence clips inside the current stage
+    to be written on executing a USD ROP.
+
+    """
+    label = "Collect USD Value Clips"
+    # Run after core plugin `CollectResourcesPath`
+    order = pyblish.api.CollectorOrder + 0.496
+    families = ["usd"]
+
+    def process(self, instance):
+
+        stage = instance.data.get("stage", None)
+        if stage is None:
+            return
+
+        prim = stage.GetPrimAtPath("/HoudiniLayerInfo")
+
+        editor_nodes = prim.GetCustomDataByKey("HoudiniEditorNodes")
+
+        # Houdini LOPs only allows one geoclip.
+        # any newer geoclip overrides the previous one.
+        clip_node = None
+        for node_id in editor_nodes:
+            node = hou.nodeBySessionId(node_id)
+            if node.type().name() == "geoclipsequence":
```

Comment:
Does this still detect the correct way if the node is inside an HDA? (e.g. part of some custom tooling) I wonder if it then still points to the internal node or not?

Reply:
Cool - I guess reliable enough for now then.